src/reducers/entities/preferences.js

```diff
@@ -10,6 +10,8 @@ function getKey(preference) {
 
 function myPreferences(state = {}, action) {
     switch (action.type) {
+    case PreferencesTypes.SAVE_PREFERENCES_OPTIMISTIC:
+    case PreferencesTypes.SAVE_PREFERENCES_ROLLBACK:
```

Comment:
Yeah, this PR is getting a major work over. I'm making the offline store the default as well as making the save and delete preferences optimistic on the redux side instead of the client side. Have it all working just need to fix the tests. Hopefully submit something soon.